js/src/tab.js

```diff
@@ -17,158 +17,262 @@ import BaseComponent from './base-component'
-      complete()
-    }
-  }
+    element.classList.add(CLASS_NAME_ACTIVE)
 
-  // Private
-  _activate(element, container, callback) {
-    const activeElements = container && (container.nodeName === 'UL' || container.nodeName === 'OL') ?
-      SelectorEngine.find(SELECTOR_ACTIVE_UL, container) :
-      SelectorEngine.children(container, SELECTOR_ACTIVE)
+    this._activate(getElementFromSelector(element)) // Search and activate/show the proper section
 
-    const active = activeElements[0]
-    const isTransitioning = callback && (active && active.classList.contains(CLASS_NAME_FADE))
+    const isAnimated = element.classList.contains(CLASS_NAME_FADE)
+    const complete = () => {
+      if (isAnimated) { // todo: maybe is redundant
+        element.classList.add(CLASS_NAME_SHOW)
+      }
 
-    const complete = () => this._transitionComplete(element, active, callback)
+      if (element.getAttribute('role') !== 'tab') {
+        return
+      }
 
-    if (active && isTransitioning) {
-      active.classList.remove(CLASS_NAME_SHOW)
-      this._queueCallback(complete, element, true)
-    } else {
-      complete()
+      element.focus()
+      element.removeAttribute('tabindex')
+      element.setAttribute('aria-selected', true)
+      this._toggleDropDown(element, true)
+      EventHandler.trigger(element, EVENT_SHOWN, {
+        relatedTarget: relatedElem
+      })
     }
+
+    this._queueCallback(complete, element, isAnimated)
   }
 
-  _transitionComplete(element, active, callback) {
-    if (active) {
-      active.classList.remove(CLASS_NAME_ACTIVE)
+  _deactivate(element, relatedElem) {
+    if (!element) {
+      return
+    }
+
+    element.classList.remove(CLASS_NAME_ACTIVE)
+    element.blur()
 
-      const dropdownChild = SelectorEngine.findOne(SELECTOR_DROPDOWN_ACTIVE_CHILD, active.parentNode)
+    this._deactivate(getElementFromSelector(element)) // Search and deactivate the shown section too
 
-      if (dropdownChild) {
-        dropdownChild.classList.remove(CLASS_NAME_ACTIVE)
+    const isAnimated = element.classList.contains(CLASS_NAME_FADE)
+    const complete = () => {
+      if (isAnimated) { // todo maybe is redundant
+        element.classList.remove(CLASS_NAME_SHOW)
       }
 
-      if (active.getAttribute('role') === 'tab') {
-        active.setAttribute('aria-selected', false)
+      if (element.getAttribute('role') !== 'tab') {
+        return
       }
+
+      element.setAttribute('aria-selected', false)
+      element.setAttribute('tabindex', '-1')
+      this._toggleDropDown(element, false)
+      EventHandler.trigger(element, EVENT_HIDDEN, { relatedTarget: relatedElem })
     }
 
-    element.classList.add(CLASS_NAME_ACTIVE)
-    if (element.getAttribute('role') === 'tab') {
-      element.setAttribute('aria-selected', true)
+    this._queueCallback(complete, element, isAnimated)
+  }
+
+  _keydown(event) {
+    if (!([ARROW_LEFT_KEY, ARROW_RIGHT_KEY].includes(event.key))) {
+      return
+    }
```

Comment:
See note 2 at the end of https://www.w3.org/TR/wai-aria-practices/#keyboard-interaction-21

> When a tab list has its [aria-orientation](https://www.w3.org/TR/wai-aria-1.2/#aria-orientation) set to vertical:
> * Down Arrow performs as Right Arrow is described above.
> * Up Arrow performs as Left Arrow is described above.

For simplicity's sake, and because assistive tech don't all consistently announce `aria-orientation` anyway (as far as the user is concerned, they're in a tab list, with no hint whether it's horizontal or vertical), I'd actually say it's pointless checking explicitly what the orientation is, and instead listening to and reacting the same to *both* down/right and up/left the same way (i.e. that you can go to previous with up/left and next with down/right regardless of the orientation)

Reply:
DONE ;)